integration-tests/src/tests/client/undo_block.rs

```diff
@@ -0,0 +1,80 @@
+use near_chain::{
+    ChainGenesis, ChainStore, ChainStoreAccess, Provenance, RuntimeWithEpochManagerAdapter,
+};
+use near_chain_configs::Genesis;
+use near_client::test_utils::TestEnv;
+use near_o11y::testonly::init_test_logger;
+use near_store::test_utils::create_test_store;
+use near_store::Store;
+use near_undo_block::undo_block;
+use nearcore::config::GenesisExt;
+use std::path::Path;
+use std::sync::Arc;
+
+/// Setup environment with one Near client for testing.
+fn setup_env(
+    genesis: &Genesis,
+    store: Store,
+) -> (TestEnv, Arc<dyn RuntimeWithEpochManagerAdapter>) {
+    let chain_genesis = ChainGenesis::new(genesis);
+    let runtimes: Vec<Arc<dyn RuntimeWithEpochManagerAdapter>> =
+        vec![nearcore::NightshadeRuntime::test(Path::new("../../../.."), store, genesis)];
+    (
+        TestEnv::builder(chain_genesis).runtime_adapters(runtimes.clone()).build(),
+        runtimes[0].clone(),
+    )
+}
+
+fn test_undo_block(epoch_length: u64, stop_height: u64) {
```

Comment:
Deserves short comment, like "Checks that Near client can successfully undo block on given height and process it again after restart"